### ISSUE TITLE
SQLite: return the number of rows changed

### ISSUE THREAD

**Issue by spiffytech** [oven-sh/bun#3284]
### What is the problem this feature would solve?

It can be useful to determine whether an `UPDATE` or `DELETE` query changed any data, and how much. I.e., did my `WHERE` actually match anything?

### What is the feature you are proposing to solve the problem?

`better-sqlite3`'s `.run()` method returns on object that knows number of rows changed. 

`pg`'s `Result` objects behave similarly.

`pg-promise` appears to offer a dedicated `db.result()` method for running a query that discards results but retrieves the count of modified rows.

### What alternatives have you considered?

Adding `RETURNING *` to the relevant queries. Works for the scenarios I've encountered so far, but no telling if it'll always be appropriate, and it makes the DB and Bun do more work than necessary.

**Comment by Jarred-Sumner** [comment]
It does this when there are no columns to return (similar to better-sqlite3) 

https://github.com/oven-sh/bun/blob/50a18b6bacd8a3e27f4ac82fb5cc38eb9a5f3029/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L1216-L1220

**Comment by spiffytech** [comment]
Ah, I see: it looks like the change count is returned by `.all()`?

Of these five ways to run a query, that's the only one that returns anything:

```ts
db.query("create table foo (id text)").run();
db.query("insert into foo (id) values (null)").run();

console.log(db.run('update foo set id = "example"'));
console.log(db.exec('update foo set id = "example"'));
console.log(db.query('update foo set id = "example"').get());
console.log(db.query('update foo set id = "example"').all());
console.log(db.query('update foo set id = "example"').values());
```

```
$ bun run foo.tsx
[0.04ms] ".env"
undefined
undefined
null
1
null
```

I don't see this mentioned in the docs, which say `.all()` returns an array of objects, and the return type for `.all()` is `unknown[]`.

If you're open to API changes, I find this behavior unintuitive. It feels odd for `.all()` to return a non-array, and to use `.all()` when I want a single-value result. Also, if the return type is updated to be `unknown[] | number`, it will either mean casting or type guarding the return types on every `.all()` query.

I just tested with `better-sqlite3`, and `.run()` is the only method that returns a changes object. `stmt.get()` and `stmt.all()` throw `TypeError: This statement does not return data. Use run() instead`.

**Comment by vitaly-t** [comment]
> `pg-promise` appears to offer a dedicated `db.result()` method for running a query that discards results but retrieves the count of modified rows.

It does not discard results, it returns complete results, an object that has `{rowCount, rows,...}`

**Comment by drag0n-app** [comment]
we need this feature

**Comment by vitaly-t** [comment]
> `pg-promise` appears to offer a dedicated `db.result()` method for running a query that discards results but retrieves the count of modified rows.

Method `result` does not discard anything, the opposite - the preserves all the original data, and even extends it. Row count is just one piece available within the `Result` object.

**Comment by MaoDoBan** [comment]
> ```ts
> db.query("create table foo (id text)").run();
> db.query("insert into foo (id) values (null)").run();
> console.log(db.query('update foo set id = "example"').all());
> ```
> 
> ```
> $ bun run foo.tsx
> [0.04ms] ".env"
> 1
> ```
> 
> I don't see this mentioned in the docs, which say `.all()` returns an array of objects, and the return type for `.all()` is `unknown[]`.
> 
> If you're open to API changes, I find this behavior unintuitive. It feels odd for `.all()` to return a non-array, and to use `.all()` when I want a single-value result. Also, if the return type is updated to be `unknown[] | number`, it will either mean casting or type guarding the return types on every `.all()` query.

They fixed on bun 1.0.7 the number returned on `.all()` inserts, now its return always an empty array `[]`:
https://github.com/oven-sh/bun/pull/5946/commits/aa46cf3e55e4a8c6b06bad8110c2d88e1d9f8136

I'm running a `SELECT changes();` query to see the previous inserted rows.

**Comment by sidesteps** [comment]
> It does this when there are no columns to return (similar to better-sqlite3)
> 
> https://github.com/oven-sh/bun/blob/50a18b6bacd8a3e27f4ac82fb5cc38eb9a5f3029/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L1216-L1220

Looks like this check for no columns just needs to be transered to 
`JSC_DEFINE_HOST_FUNCTION(jsSQLStatementExecuteStatementFunctionGet...`
from 
`JSC_DEFINE_HOST_FUNCTION(jsSQLStatementExecuteStatementFunctionAll...`

to fix this.
